src/main/kotlin/it/pagopa/ecommerce/cdc/services/TransactionViewUpsertService.kt

```diff
@@ -180,53 +268,213 @@ class TransactionViewUpsertService(
+        statusUpdate["status"] = TransactionStatusDto.CLOSURE_REQUESTED
+        statusUpdate["lastProcessedEventAt"] =
+            ZonedDateTime.parse(event.creationDate).toInstant().toEpochMilli()
+        return Pair(null, statusUpdate)
     }
 
     /**
      * Updates fields for TRANSACTION_USER_RECEIPT_REQUESTED_EVENT. Adds user receipt information.
      */
-    private fun updateUserReceiptData(event: TransactionUserReceiptRequestedEvent): Update {
+    private fun updateUserReceiptData(
+        event: TransactionUserReceiptRequestedEvent
+    ): Pair<Update?, Update> {
         val update = Update()
-        update["sendPaymentResultOutcome"] = event.data.responseOutcome.toString()
-        return update
+        val statusUpdate = Update()
+        update["sendPaymentResultOutcome"] = event.data.responseOutcome
+        statusUpdate["sendPaymentResultOutcome"] = event.data.responseOutcome
+
+        statusUpdate["status"] = TransactionStatusDto.NOTIFICATION_REQUESTED
+        statusUpdate["lastProcessedEventAt"] =
+            ZonedDateTime.parse(event.creationDate).toInstant().toEpochMilli()
+
+        return Pair(update, statusUpdate)
+    }
+
+    /** Updates fields for TRANSACTION_EXPIRED_EVENT. Adds expiration information. */
+    private fun updateExpiredData(event: TransactionExpiredEvent): Pair<Update?, Update> {
+        val statusUpdate = Update()
+        statusUpdate["status"] = TransactionStatusDto.EXPIRED
+        statusUpdate["lastProcessedEventAt"] =
+            ZonedDateTime.parse(event.creationDate).toInstant().toEpochMilli()
+        return Pair(null, statusUpdate)
+    }
+
+    /** Updates fields for TRANSACTION_REFUND_REQUESTED_EVENT. Adds refund request information. */
+    private fun updateRefundRequestData(
+        event: TransactionRefundRequestedEvent
+    ): Pair<Update?, Update> {
+        val statusUpdate = Update()
+        statusUpdate["status"] = TransactionStatusDto.REFUND_REQUESTED
+        statusUpdate["lastProcessedEventAt"] =
+            ZonedDateTime.parse(event.creationDate).toInstant().toEpochMilli()
+        return Pair(null, statusUpdate)
     }
 
     /**
      * Updates fields for TRANSACTION_CLOSED_EVENT. Sets sendPaymentResultOutcome and closure
      * timestamp.
      */
-    private fun updateClosedData(event: TransactionClosedEvent): Update? {
-        val update = Update()
-        return if (event.data.responseOutcome == TransactionClosureData.Outcome.OK) {
-            update["sendPaymentResultOutcome"] =
-                TransactionUserReceiptData.Outcome.NOT_RECEIVED.toString()
-            update["closureErrorData"] = null
-            update
-        } else {
-            null
-        }
+    private fun updateClosedData(event: TransactionClosedEvent): Pair<Update?, Update> {
+        val statusUpdate = Update()
+        statusUpdate["sendPaymentResultOutcome"] = TransactionUserReceiptData.Outcome.NOT_RECEIVED
+        statusUpdate["closureErrorData"] = null
```

Comment:
why linking this update with status only and not duplicating it as done for others?